src/main/kotlin/com/ajaxproject/telegrambot/bot/repository/CurrencyExchangeRepositoryImpl.kt

```diff
@@ -0,0 +1,41 @@
+package com.ajaxproject.telegrambot.bot.repository
+
+import com.ajaxproject.telegrambot.bot.enums.Currency
+import com.ajaxproject.telegrambot.bot.models.currency.MongoCurrency
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.stereotype.Repository
+
+@Repository
+class CurrencyExchangeRepositoryImpl(
+    private val mongoTemplate: MongoTemplate,
+) : CurrencyExchangeRepository {
+    override fun findByCode(code: Int): List<MongoCurrency> {
+        val query: Query = Query().addCriteria(
+            Criteria.where("currency_code_A").`is`(code)
+                .orOperator(
+                    Criteria.where("currency_code_B").`is`(Currency.UAH.code),
+                    Criteria.where("currency_code_B").`is`(Currency.USD.code),
+                    Criteria.where("currency_code_B").`is`(Currency.EUR.code)
```

Comment:
https://www.mongodb.com/docs/manual/reference/operator/query/in/

Let's simplify your query with `in` operator

Reply:
Note, you can use `Currency.values.map { it.code }` to enumerate all values if needed